nativelib/src/main/resources/scala-native/gc/info/GCInfo.h

```diff
@@ -0,0 +1,9 @@
+#ifndef GC_INFO_H
+#define GC_INFO_H
+
+#define GC_TYPE_IMMIX "immix"
```

Comment:
I also wanted to include a feature like this since Boehm apparently has an option for setting finalize functions, so we could potentially replace setting handlers in WeakReferenceRegistry with that if needed (and therefore supporting pseudo-finalization with all 3 of our supported GCs). I suppose this can be solved with link time conditions as well. I'll try to look into this